gqrx-scan.c

```diff
@@ -1580,6 +1571,16 @@ int main(int argc, char **argv) {
         print_usage(argv[0]);
     }
 
+    if (opt_scan_mode == sweep)
+    {
+        size_t freqeuencies_count = (size_t)(((opt_max_freq-opt_min_freq)/opt_scan_bw)+1);
```

Comment:
This could grow very very big in case of large scan. Do you really need to store the noise floor for each step? I was even tempted to remove the GetSquelchLevel from the inner loop (to speed up a bit) and get a value before each round (the user will not change the setting during the scan, usually, so it is not optimized). Can we do something like that to avoid the fine-grained array?

Reply:
> This could grow very very big in case of large scan. Do you really need to store the noise floor for each step? I was even tempted to remove the GetSquelchLevel from the inner loop (to speed up a bit) and get a value before each round (the user will not change the setting during the scan, usually, so it is not optimized). Can we do something like that to avoid the fine-grained array?

We need to know, for each frequency what noise floor we are dealing with and yeah - each one. If we have ie. UKF NFM 12500 Hz with bandwidth 8500 Hz or USB which can be even narrower and we would mesh it more loosely we would end up with omitting chatter.
In the other hand: lets say 2MHz scanning every 12500 Hz we are ending up with 160 elements, 1856 bytes each (if I remember correctly) we are ending up with 290 KiB of memory space. C'mon ( ͡° ͜ʖ ͡°) it's not that much.
If we go crazy and we wanted to cover whole 800MHz we would end up with 113 MiB.